aiida/cmdline/params/options/commands/code.py

```diff
@@ -68,11 +80,39 @@ def validate_label_uniqueness(ctx, _, value):
     return value
 
 
+ON_CONTAINER = OverridableOption(
+    '--on-container/--not-on-container',
+    is_eager=False,
+    default=False,
+    help='Whether the code is installed on the container.'
+)
+
+IMAGE = OverridableOption(
+    '--image',
+    prompt='image name',
+    required_fn=is_on_container,
+    prompt_fn=is_on_container,
+    cls=InteractiveOption,
+    help='image name.'
+)
+
+CONTAINER_CMDLINE_TMPL = OverridableOption(
+    '--container-cmdline-tmpl',
+    prompt='container cmdline params tmpl',
+    default='sarus run --mount=src=$PWD,dst=/workir,type=bind --workdir /workdir {image}',
+    required_fn=is_on_container,
+    prompt_fn=is_on_container,
+    cls=InteractiveOption,
+    help='container cmdline params tmpl.'
+)
+
 ON_COMPUTER = OverridableOption(
     '--on-computer/--store-in-db',
     is_eager=False,
     default=True,
     cls=InteractiveOption,
+    required_fn=is_not_on_container,
```

Comment:
hm... why?

Reply:
What is this `in-db`? it is the other way to say for the local code, no? So either we on_computer or 'not_on_computer' which is local (in-db?) or we on_container?